app/controllers/admin/stores_controller.rb

```diff
@@ -7,7 +7,7 @@ def index
 
   def update
     store = Store.find(params[:id])
-    store.update_status(params[:status])
+    store.update_status(params[:status], params[:initial_req])
```

Comment:
Unless the store's status is being changed for an initial request, "params[:initial_req]" will have the value of nil.